File: domain/src/main/java/io/prizy/domain/contest/ports/PackRepository.java
package io.prizy.domain.contest.ports;

import java.util.Optional;
import java.util.UUID;

import io.prizy.domain.contest.model.Pack;

/**
 * @author Nidhal Dogga
 * @created 4/24/2022 1:02 AM
 */

public interface PackRepository {
  Optional<Pack> byId(UUID id);
}

File: domain/src/main/kotlin/io/prizy/domain/contest/ports/PackRepository.kt
package io.prizy.domain.contest.ports

import io.prizy.domain.contest.model.Pack
import java.util.UUID

/**
 *  @author Nidhal Dogga
 *  @created 4/24/2022 1:02 AM
 */

interface PackRepository {
  suspend fun byId(id: UUID): Pack?
}
